controller-server/src/main/java/com/yahoo/vespa/hosted/controller/role/Context.java

```diff
@@ -0,0 +1,74 @@
+// Copyright 2019 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.hosted.controller.role;
+
+import com.yahoo.config.provision.ApplicationName;
+import com.yahoo.config.provision.SystemName;
+import com.yahoo.config.provision.TenantName;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * The context in which a role is valid.
+ *
+ * @author mpolden
+ */
+public class Context {
```

Comment:
The presence of a tenant or application name is a restriction on the context — while absence is a wildcard entry. I think this should be clearly reflected in the naming of at least the factory methods.

Reply:
E.g., `empty()` should be `allIn()`, or something similar — preferably something better.